src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -781,4 +798,35 @@ public function hasAutowiringType($type)
 
         return isset($this->autowiringTypes[$type]);
     }
+
+    /**
+     * Returns all changes tracked for the Definition object.
+     *
+     * @return array An array of changes for this Definition
+     */
+    public function getChanges()
+    {
+        return $this->changes;
+    }
+
+    /**
+     * Turn internal change tracking on or off.
+     *
+     * @param bool $trackChanges
+     *
+     * @return $this
+     */
+    public function setTrackChanges($trackChanges)
```

Comment:
What about marking this as `@internal` ? I don't think we should bother about untracked changes due to a choice of the user.
Maybe we could even use a bound callable where necessary.